src/Effect/Grapher.hs

```diff
@@ -49,6 +50,7 @@ data SGrapher ty k where
   Direct :: ty -> SGrapher ty ()
   Edge :: ty -> ty -> SGrapher ty ()
   Deep :: ty -> SGrapher ty ()
+  HasVertex :: ty -> SGrapher ty Bool
```

Comment:
Can we also add a `HasEdge` helper here?  I think it make sense to do both, and it may help with other strategies down the line.

Reply:
I've gone ahead and added this.